src/main/java/com/tileman/TilemanModePlugin.java

```diff
@@ -410,17 +429,147 @@ private void handleWalkedToTile(LocalPoint currentPlayerPoint) {
         // TODO Fill path between last point and current point. This will fix missing tiles that occur when you lag
         // TODO   and rendered frames are skipped. See if RL has an api that mimic's OSRS's pathing. If so, use that to
         // TODO   set all tiles between current tile and lastTile as marked
-        if (lastTile != null
-                && (lastTile.distanceTo(currentPlayerPoint) == 256 || lastTile.distanceTo(currentPlayerPoint) == 362)) {
-            int xDiff = lastTile.getX() - currentPlayerPoint.getX();
-            int yDiff = lastTile.getY() - currentPlayerPoint.getY();
+        if(lastTile != null){
+            int xDiff = currentPlayerPoint.getX() - lastTile.getX();
+            int yDiff = currentPlayerPoint.getY() - lastTile.getY();
             int yModifier = yDiff / 2;
             int xModifier = xDiff / 2;
 
-            updateTileMark(new LocalPoint(currentPlayerPoint.getX() + xModifier, currentPlayerPoint.getY() + yModifier), true);
+            switch(lastTile.distanceTo(currentPlayerPoint)) {
+                case 0:
+                    return;
+                case 256:
+                case 362:
+                    updateTileMark(new LocalPoint(lastTile.getX() + xModifier, lastTile.getY() + yModifier), true);
+                    break;
+                case 286:
+                    handleLMovement(xDiff, yDiff);
+                    break;
+                case 181:
+                    handleCornerMovement(xDiff, yDiff);
+                    break;
+            }
+        }
+    }
+
+    private void handleLMovement(int xDiff, int yDiff) {
+        int xModifier = xDiff / 2;
+        int yModifier = yDiff / 2;
+        int tileBesideXDiff, tileBesideYDiff;
+
+        // Whichever direction has moved only one, keep it 0. This is the translation to the potential 'problem' gameObject
+        if (Math.abs(yDiff) == 128) {
+            tileBesideXDiff = xDiff;
+            tileBesideYDiff = 0;
+        } else {
+            tileBesideXDiff = 0;
+            tileBesideYDiff = yDiff;
+        }
+
+        MovementFlag[] tileBesideFlagsArray = getTileMovementFlags(lastTile.getX() + tileBesideXDiff, lastTile.getY() + tileBesideYDiff);
+
+        if (tileBesideFlagsArray.length == 0) {
+            updateTileMark(new LocalPoint(lastTile.getX() + tileBesideXDiff / 2, lastTile.getY() + tileBesideYDiff / 2), true);
+        } else if (containsAnyOf(tileBesideFlagsArray, fullBlock) || !containsAnyOf(allDirections, tileBesideFlagsArray)) {
+            if (yModifier == 64) {
+                yModifier = 128;
+            } else if (xModifier == 64) {
+                xModifier = 128;
+            }
+            updateTileMark(new LocalPoint(lastTile.getX() + xModifier, lastTile.getY() + yModifier), true);
+        } else {
```

Comment:
can we add a comment describing which case this is handling?

Reply:
Changed the logic, should be more readable now without comment